sunpy/map/tests/test_mapbase.py

```diff
@@ -1890,3 +1891,22 @@ def test_submap_nan_error(aia171_test_map):
     coord_other = SkyCoord(0*u.arcsec, 0*u.arcsec, frame='helioprojective', observer='earth', obstime=aia171_test_map.date)
     with pytest.raises(ValueError, match="The provided input coordinates to"):
         aia171_test_map.submap(coord_other, width=1000*u.arcsec, height=1000*u.arcsec)
+
+
+def test_slicing_map_submap_same(aia171_test_map):
+    bl_coord = SkyCoord(-1000 * u.arcsec, 0 * u.arcsec, frame=aia171_test_map.coordinate_frame)
+    bottom_left = aia171_test_map.wcs.world_to_pixel(bl_coord)
+
+    tp_coord = SkyCoord(0 * u.arcsec, 1000 * u.arcsec, frame=aia171_test_map.coordinate_frame)
+    top_right = aia171_test_map.wcs.world_to_pixel(tp_coord)
+
+    bottom_left_x, bottom_left_y = map(int, bottom_left)
+    top_right_x, top_right_y = map(int, top_right)
+
+    submap = aia171_test_map.submap([bottom_left_x, bottom_left_y]*u.pix,
+                                    top_right=[top_right_x, top_right_y]*u.pix
+                                    )
+
+    sliced_map = aia171_test_map[bottom_left_y : top_right_y + 1, bottom_left_x: top_right_x + 1]
+
+    assert_cubes_equal(submap, sliced_map)
```

Comment:
So the original map and the submap lack the meta keys:

'wcsaxes', 'pc1_1', 'pc1_2', 'pc2_1', 'pc2_2', 'lonpole', 'latpole', 'mjdref', 'mjd-obs', 'date-avg', 'mjd-avg'

whereas the sliced_map has them and it fails on this check.

Reply:
Also cunit has changed from arcsecs to degrees for the sliced_map.